blockchain/blockdao/blockdao.go

```diff
@@ -216,6 +252,43 @@ func (dao *blockDAO) initStores() error {
 	return nil
 }
 
+func (dao *blockDAO) StartExistingBlockchain(ctx context.Context, gas uint64) error {
+	if dao.sf == nil {
+		return errors.New("statefactory cannot be nil")
+	}
+	stateHeight, err := dao.sf.Height()
+	if err != nil {
+		return err
+	}
+	tipHeight := dao.GetTipHeight()
+	if stateHeight > tipHeight {
+		return errors.New("factory is higher than blockchain")
+	}
+
+	for i := stateHeight + 1; i <= tipHeight; i++ {
+		blk, err := dao.GetBlockByHeight(i)
+		if err != nil {
+			return err
+		}
+		producer, err := address.FromBytes(blk.PublicKey().Hash())
+		if err != nil {
+			return err
+		}
+		ctx = dao.contextWithBlock(ctx, producer, blk.Height(), blk.Timestamp(), gas)
+		if err := dao.sf.PutBlock(ctx, blk); err != nil {
+			return err
+		}
+	}
```

Comment:
this is something sf should do by itself

Reply:
have to copy all GetBlockByHeight related function to sf,I think it's not necessary